docs/generated_code.cpp

```diff
@@ -0,0 +1,347 @@
+        auto oit=ot.require_find(oracle.value,"?"); ot.modify(oit,same_payer,[&](auto&r){
+            if(++r.strikes>=_conf().strikes_max) r.suspended=true;});
+        return;
+    }
+    checksum256 rnd=sha256(sig.data(),384);
+    action{ {get_self(),"active"_n}, rit->dapp,"receiverand"_n,
+            std::make_tuple(rnd) }.send();
+
+    _reward_oracles(_conf().fee_per_call);
+    rt.erase(rit);
+}
+
+/* reward split */
+void orng::_reward_oracles(asset qty){
+    oracles_table ot(get_self(),get_self().value);
+    if(ot.empty()) return;
+    asset each{ qty.amount/static_cast<int64_t>(ot.size()), WAX };
+    bal_table bt(get_self(),get_self().value);
+    for(auto& o:ot){
+        auto it=bt.find(o.oracle.value);
+        if(it==bt.end()) bt.emplace(get_self(),[&](auto&r){ r.oracle=o.oracle;r.unpaid=each;});
+        else bt.modify(it,same_payer,[&](auto&r){ r.unpaid+=each;});
+    }
+}
+void orng::claim(name o){
+    _ensure_not_paused(); require_auth(o);
+    bal_table bt(get_self(),get_self().value);
+    auto it=bt.require_find(o.value,"none"); check(it->unpaid.amount>0,"zero");
+    asset pay=it->unpaid; bt.modify(it,same_payer,[&](auto&r){ r.unpaid.amount=0;});
+    action{ {get_self(),"active"_n},TOKEN,"transfer"_n,
+            std::make_tuple(get_self(),o,pay,string("rng reward")) }.send();
+}
+
+/* dapperror */
+void orng::dapperror(name d,uint64_t assoc,std::string m){
+    require_auth(permission_level{d,"ornglog"_n});
+    err_table et(get_self(),d.value);
+    while(et.size()>19) et.erase(et.begin());
+    et.emplace(d,[&](auto&r){
+        r.id=et.available_primary_key(); r.assoc=assoc; r.msg=m; r.logged=current_time_point();
+    });
+}
+
+/* governance small ones */
+void orng::ban(name d){ require_auth(GOV);
+    ban_table bt(get_self(),get_self().value); check(bt.find(d.value)==bt.end(),"dup");
+    bt.emplace(get_self(),[&](auto&r){ r.dapp=d;});
+}
+void orng::unban(name d){ require_auth(GOV);
+    ban_table bt(get_self(),get_self().value);
+    auto it=bt.require_find(d.value,"?"); bt.erase(it);
+}
+void orng::pause(bool p){ require_auth(GOV); auto c=_conf(); c.paused_all=p;
+    config_singleton(get_self(),get_self().value).set(c,get_self()); }
+void orng::pauserequest(bool p){ require_auth(GOV); auto c=_conf(); c.paused_req=p;
+    config_singleton(get_self(),get_self().value).set(c,get_self()); }
+
+void orng::setconfig(asset f,uint8_t sm,uint8_t k){
+    require_auth(GOV); check(f.symbol==WAX,"sym");
+    auto c=_conf(); c.fee_per_call=f; c.strikes_max=sm; c.k_calls_per_wax=k;
+    config_singleton(get_self(),get_self().value).set(c,get_self());
+}
+void orng::setpubkey(uint8_t ver,checksum256 mod,uint32_t exp){
+    require_auth(GOV); pkey_table pk(get_self(),get_self().value);
+    pk.emplace(get_self(),[&](auto&r){ r.ver=ver;r.modulus=mod;r.exponent=exp;});
+    auto c=_conf(); c.active_ver=ver;
+    config_singleton(get_self(),get_self().value).set(c,get_self());
```

Comment:
I think we are supposed to set the previously active public key to retired here. It will be useful for eyeballing the public keys for retired status